workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.workers.common
+
+import org.eclipse.apoapsis.ortserver.model.OrtRun
+
+import org.ossreviewtoolkit.model.OrtResult
+
+/**
+ * An internally used helper class to generate an [OrtResult] from the current [OrtRun].
+ *
+ * This class loads the results of the previous worker steps and generates an [OrtResult] from them. In addition, it
+ * adds a number of common labels that can be evaluated by different types of workers to obtain further information
+ * about the run.
+ *
+ * The following labels are added:
+ * - runId: The id of the [OrtRun].
+ */
+object OrtBaseResultGenerator {
+    private const val RUN_ID_LABEL: String = "runId"
+
+    fun generate(ortRunService: OrtRunService, ortRun: OrtRun, failIfRepoInfoMissing: Boolean = true): OrtResult {
```

Comment:
The function could support a map parameter for additional labels to be added which defaults to an empty map. This would simplify its usage for clients; they would then not have to bother with copying the result and adding further labels.

Reply:
Please let's have a separate feature request for that.

Reply:
This was not meant as a new feature request, but as a means to simplify the current implementation - the result generator for the notifier would directly benefit from it. But it's fine with me, I won't insist.